x-pack/plugins/infra/public/components/metrics_explorer/chart_context_menu.test.tsx

```diff
@@ -82,4 +93,41 @@ describe('MetricsExplorerChartContextMenu', () => {
     expect(menuItems.length).toBe(1);
     expect(menuItems.at(0).prop('disabled')).toBeTruthy();
   });
+
+  it('should not display "Open in Visualize" when unavailble in uiCapabilities', async () => {
+    const customUICapabilities = { ...uiCapabilities, visualize: { show: false } };
+    const onFilter = jest.fn().mockImplementation((query: string) => void 0);
+    const component = mountWithIntl(
+      <MetricsExplorerChartContextMenu
+        timeRange={timeRange}
+        source={source}
+        series={series}
+        options={options}
+        onFilter={onFilter}
+        uiCapabilities={customUICapabilities}
+      />
+    );
+
+    component.find('button').simulate('click');
+    const menuItems = component.find('button.euiContextMenuItem');
```

Comment:
Maybe we could avoid relying on internal EUI class names by using `data-test-subj` also in jest tests? That would also make it easier to check for non-existence as I mentioned in the above comment.

Reply:
coming with #37136